packages/bigquery/sodasql/dialects/bigquery_dialect.py

```diff
@@ -50,6 +50,15 @@ def default_connection_properties(self, params: dict):
             'dataset': params.get('database', 'Eg your_bigquery_dataset')
         }
 
+    def safe_connection_data(self):
+        return [
+            self.type,
+            self.dataset_name,
+            self.account_info_dict['type'] or None,
+            self.account_info_dict['project_id'] or None,
```

Comment:
`project_id` is what we want to push to `datasource_id` for BQ

Reply:
I wasnt sure whether project id is globally unique so added more attributes, but now I found this

> Each project ID is unique across Google Cloud. Once you have created a project, you can delete the project but its ID can never be used again.
(https://cloud.google.com/docs/overview?hl=id) 

so yes it is safe to only use project id, changed

Note - we will not be sending the actual project id as `datasource_id`, but the hash of it.